File: PackWize/commands/migrate.py
from PackWize.utils.get_modpack_dir import get_modpack_dir
from PackWize.utils.run_cmd import run_cmd

def migrate(minecraft_versions: list[str], launchers: list[str], target: str, version: str) -> None:
    """
    Migrate modpacks to a different format for multiple Minecraft versions and launchers.

    Arguments:
        minecraft_versions: list[str]. List of Minecraft versions
        launchers: list[str]. List of launchers
        target: str. The target format to migrate to (e.g., "modrinth", "curseforge")
        version: str. The version of the target format to migrate to (e.g., "1.0.0")
    
    Returns nothing
    """
    for minecraft_version in minecraft_versions:
        for launcher in launchers:
            modpack_dir = get_modpack_dir(minecraft_version, launcher)

            run_cmd(['packwiz', 'migrate', target, version], modpack_dir)

File: pkg/mods/migrate.go
package mods

import (
	"fmt"
	"log"
	"os"
	"os/exec"

	"github.com/Alfakynz/PackWize/pkg/utils"
)

// MigrateModpack migrates a modpack to another Minecraft or loader version
func MigrateModpack(minecraftVersionArg, launcherArg, target, version string) {
	// Convert arguments
	versions := utils.ConvertArguments("minecraft_versions", minecraftVersionArg)
	launchers := utils.ConvertArguments("launchers", launcherArg)

	if versions == nil || launchers == nil {
		log.Printf("Invalid Arguments: %s / %s\n", minecraftVersionArg, launcherArg)
		return
	}

	// Loop over all combinations
	for _, v := range versions {
		for _, l := range launchers {
			fmt.Printf("Migrating the %s version to %s in %s/%s ...\n", target, version, v, l)

			cmd := exec.Command("packwiz", "migrate", target, version)
			cmd.Dir = fmt.Sprintf("%s/%s", v, l)
			cmd.Stdout = os.Stdout
			cmd.Stderr = os.Stderr
			cmd.Stdin = os.Stdin

			if err := cmd.Run(); err != nil {
				log.Printf("Error migrating the modpack in %s/%s: %v\n", v, l, err)
			}
		}
	}
}